fitbenchmarking/fitting/sasview/main.py

```diff
@@ -0,0 +1,171 @@
+
+        results_problem.append(individual_result)
+
+    return results_problem, best_fit
+
+def fit(problem, data, function, minimizer, init_func_def):
+    """
+
+    @param problem :: a problem object containing information used in fitting
+    @param model :: the fitted function (model for SasView problems)
+    @param minimizer :: the minimizer used in the fitting
+    @param init_func_def :: the initial function definition
+
+    @returns ::  the status, either success or failure (str), the data
+                of the fit, the final function definition and the
+                runtime of the fitting software
+    """
+
+    t_start, t_end = None, None
+    model = function[0]
+
+    if hasattr(model, '__call__'):
+        if isinstance(problem.starting_values, basestring):
+            function_without_ties = re.sub(r",(\s+)?ties=[(][A-Za-z0-9=.,\s+]+[)]", '', problem.equation)
+            function_list = (function_without_ties).split(';')
+            func_params_list = [(func.split(','))[1:] for func in function_list]
+            formatted_param_list = ['f'+str(func_params_list.index(func_params))+'.'+param.strip() for func_params in func_params_list for param in func_params]
+            param_names = [(param.split('='))[0] for param in formatted_param_list if not 'BinWidth' in param]
+            formatted_param_names = [param.replace('.', '_') for param in param_names]
+        else:
+            formatted_param_names = [param[0] for param in problem.starting_values]
+
+        param_values = function[1]
+        param_string = ''
+        for name, value in zip(formatted_param_names, param_values):
+            if not name.endswith('BinWidth'):
+                param_string += "," + name + "=" + str(value)
+
+        exec ('func_wrapper = Curve(model, x=data.x, y=data.y, dy=data.dy' + param_string + ')')
+
+        for name, value in zip(formatted_param_names, param_values):
+            minVal = -np.inf
+            maxVal = np.inf
+            if not name.endswith('BinWidth'):
+                exec ('func_wrapper.' + name + '.range(' + str(minVal) + ',' + str(maxVal) + ')')
+    else:
+        exec ("params = dict(" + problem.starting_values + ")") in locals()
+
+        model_wrapper = Model(model, **params)
+        for range in problem.starting_value_ranges.split(';'):
+            exec ('model_wrapper.' + range)
+        func_wrapper = Experiment(data=data, model=model_wrapper)
+
+    fitProblem = FitProblem(func_wrapper)
+
+    try:
+        t_start = time.clock()
+        result = bumpsFit(fitProblem, method=minimizer)
+        t_end = time.clock()
+    except (RuntimeError, ValueError) as err:
+        logger.warning("Fit failed: " + str(err))
+
+    status = 'success' if result.success is True else 'failed'
+
+    fitted_y = func_wrapper.theory()
+
+    final_param_values = result.x
+
+    fin_func_def = get_fin_function_def(final_param_values, problem, init_func_def)
```

Comment:
what does 'fin' in fin_func_def stand for

Reply:
The word fin stands for final in this case.